### PR TITLE
[Imprve][Connector-V2][Hive] Support read text table & Column projection

### LINKED ISSUE #3844: [Feature][Connector-V2][Hive] Support read text table
sub-task of #3699

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #3844 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/source/HdfsFile.md [hunk 1]
```diff
@@ -219,6 +220,20 @@ The keytab path of kerberos
 
 the schema fields of upstream data
 
+### read_columns [list]
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/HdfsFile.md:223]: add change log for this pr.

File: docs/en/connector-v2/source/LocalFile.md [hunk 2]
```diff
@@ -199,6 +200,20 @@ connector will generate data as the following:
 
 The schema information of upstream data.
 
+### read_columns [list]
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/LocalFile.md:203]: same as above.

**Review by EricJoy2048 — APPROVED** [review]

File: docs/en/connector-v2/source/HdfsFile.md (hunk 1 above)
Comment (TyrantLucifer) [reply docs/en/connector-v2/source/HdfsFile.md:223]: image

File: docs/en/connector-v2/source/LocalFile.md (hunk 2 above)
Comment (TyrantLucifer) [reply docs/en/connector-v2/source/LocalFile.md:203]: image

**Review by dijiekstra — APPROVED** [review]
LGTM

**Pull request merged**